### PR TITLE
[FormRecognizer] Test fix and sample reordering

### LINKED ISSUE #11493: Make sure sample test run and pass in our nightly builds
Currently they are not. 
Samples are located at https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples

### PR DESCRIPTION
**Pull request by kinelski**
Set of changes:
- Fixed `Ignored` sample tests.
- Moved common functionality from `FormRecognizerClientLiveTests` and `FormTrainingClientLiveTests` to the new class `FormRecognizerLiveTestBase`.
- Reordered samples to make training appear before custom recognition.
- "Train Model with Forms" and "Train Model with Forms and Labels" samples numbering updated to be the same.

Fixes #11493

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/README.md [hunk 1]
```diff
@@ -173,33 +173,10 @@ foreach (FormPage page in formPages.Value)
...
-        Console.WriteLine($"    Confidence: '{field.Confidence}");
-    }
-}
-```
-
 ### Train a Model
 Train a machine-learned model on your own form types. The resulting model will be able to recognize values from the types of forms it was trained on.
 
-```C# Snippet:FormRecognizerSample4TrainModelWithForms
+```C# Snippet:FormRecognizerSample3TrainModelWithForms
```
Comment (annelo-msft) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/README.md:179]: @maririos to discuss the current sample ordering.

**Comment by annelo-msft** [comment]
We worked with the other languages decide which samples to expose in what order.  I don't think it make sense to change these around and realign with other languages this close to the next release.

**Comment by maririos** [comment]
>Reordered samples to make training appear before custom recognition.

The agreement across languages was to leave custom before training. Users will use custom recognition more often than training. Also, we are expecting for the majority to use the Labeling Tool.

>"Train Model with Forms" and "Train Model with Forms and Labels" samples numbering updated to be the same.

Numbering goes according to complexity in the sample. I left them with different numbers because "with forms and labels" implies that you need extra documents with the labeling format and therefore is a way more complicated scenario

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Sample4_RecognizeCustomFormsFromUri.cs [hunk 2]
```diff
@@ -14,18 +15,25 @@ namespace Azure.AI.FormRecognizer.Samples
...
     {
         [Test]
-        [Ignore("Need to revisit how to pass the modelId. Issue https://github.com/Azure/azure-sdk-for-net/issues/11493")]
         public async Task RecognizeCustomFormsFromUri()
         {
             string endpoint = TestEnvironment.Endpoint;
             string apiKey = TestEnvironment.ApiKey;
+            string trainingFileUrl = TestEnvironment.BlobContainerSasUrl;
+
+            // Firstly, create a trained model we can use to recognize the custom form.
```
Comment (maririos) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Sample4_RecognizeCustomFormsFromUri.cs:24]: we should add information here about the labeling tool.
like i.e. https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs#L441

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Sample4_RecognizeCustomFormsFromUri.cs [hunk 3]
```diff
@@ -14,18 +15,25 @@ namespace Azure.AI.FormRecognizer.Samples
...
+            FormTrainingClient trainingClient = new FormTrainingClient(new Uri(endpoint), new AzureKeyCredential(apiKey));
+            CustomFormModel model = await trainingClient.StartTraining(new Uri(trainingFileUrl)).WaitForCompletionAsync();
+
+            // Proceed with the custom form recognition.
 
             FormRecognizerClient client = new FormRecognizerClient(new Uri(endpoint), new AzureKeyCredential(apiKey));
 
             string formUri = FormRecognizerTestEnvironment.CreateUri("Form_1.jpg");
-            string modelId = "<your model id>";
+            string modelId = model.ModelId;
```
Comment (maririos) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Sample4_RecognizeCustomFormsFromUri.cs:34]: we could actually include the `modelId` into the snippet, similar to what we do here: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/SampleSnippets.cs#L22

by setting `//@@ string modelId = "<modelId >";`

**Comment by kinelski** [comment]
> We worked with the other languages decide which samples to expose in what order.



> The agreement across languages was to leave custom before training.

I didn't work directly with samples in the last preview, so I wasn't aware it was an agreement across languages. I'm not really fond of the `custom` => `training` order, but I agree that deviating from other languages at this point would not be profitable. I'll revert it back.

**Comment by kinelski** [comment]
> Numbering goes according to complexity in the sample. I left them with different numbers because "with forms and labels" implies that you need extra documents with the labeling format and therefore is a way more complicated scenario

I was thinking more about the "code complexity" than the "scenario complexity" itself, since the code of the samples are basically the same, but I don't really mind that much. I think your approach is reasonable as well.

I was a bit concerned about having a `Sample4_TrainModel.md` file for both `Sample4_TrainModelWithForms.cs` and `Sample5_TrainModelWithFormsAndLabels.cs`,

while having `Sample5_ManageCustomModels.md` for `Sample6_ManageCustomModels.cs`, but maybe I'm just overconcerning.

I'll revert it back since it's not really this PR's focus. We can always discuss it again if we think it's a reason for concern.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Sample4_RecognizeCustomFormsFromUri.cs (hunk 3 above)
Comment (kinelski) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Sample4_RecognizeCustomFormsFromUri.cs:34]: Ohh, this is nice. Added it to the `FormRecognizerRecognizeCustomFormsFromFile` snippet as well.

**Review by maririos — APPROVED** [review]

**Pull request merged**